src/pages/GpuCredit/index.tsx

```diff
@@ -0,0 +1,362 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+import cn from 'classnames/bind';
+import { Bar } from 'react-chartjs-2';
+import { CategoryScale, BarElement } from 'chart.js';
+import Chart from 'chart.js/auto';
+
+import Content from 'layouts/Content';
+import { ReactComponent as JumpIcon } from 'assets/icons/jump.svg';
+import { ReactComponent as GitHubIcon } from 'assets/icons/github.svg';
+import { ReactComponent as NoCreditUsageHistoryIcon } from 'assets/images/no-credit-usage-history.svg';
+import { ReactComponent as AddIcon } from 'assets/icons/Add.svg';
+import { ReactComponent as TimeIcon } from 'assets/icons/time.svg';
+import { ReactComponent as KeyIcon } from 'assets/icons/key.svg';
+import JupyterHubImg from 'assets/images/jupyterhub.png';
+import { Table, TableHeaderProps } from 'components/Table';
+import { Button } from 'components/Button';
+import { handleConnectGithub } from 'components/GithubConnect/helper';
+import { RootState } from 'store/configure';
+import axios, { axiosAuth } from 'rest/request';
+import styles from './index.module.scss';
+
+const cx = cn.bind(styles);
+Chart.register(CategoryScale, BarElement);
+const baseApiUrl = process.env.REACT_APP_BASE_GPU_API_URL;
+
+const GpuCredit: React.FC<{}> = () => {
+  const [gpuStatistics, setGpuStatistics] = useState({
+    totalCards: 12,
+    totalVRAM: 295,
+    totalUsage: 0
+  });
+  const [gpuStatus, setGpuStatus] = useState([]);
+  const [creditUsageHistoryData, setCreditUsageHistoryData] = useState([]);
+  const [dailyCreditUsage, setDailyCreditUsage] = useState([]);
+
+  const tokens = useSelector((state: RootState) => state.auth.token);
+  const credit = useSelector((state: RootState) => state.auth.credit);
+  const loggedIn = !!tokens.access;
+
+  const [chartData, setChartData] = useState({
+    labels: dailyCreditUsage.map((data) => new Date(data.date).getDate()),
+    datasets: [
+      {
+        // label: undefined,
+        data: dailyCreditUsage.map((data) => data.credit),
+        backgroundColor: ['#7332E7'],
+        borderColor: '#B999F3',
+        borderWidth: 2,
+        borderRadius: 4,
+        borderSkipped: false,
+        barThickness: 20
+      }
+    ]
+  });
+
+  useEffect(() => {
+    if (loggedIn) {
+      const getData = async () => {
+        let promises;
+
+        try {
+          promises = await Promise.all([
```

Comment:
Can you optimize this paragraph? 

example: const [creditHistory, creditDay] = await Promise.all .....